grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/GrpcClientFactory.java

```diff
@@ -133,15 +135,15 @@ public Object newClient(ClientBuilderParams params) {
             jsonMarshaller = null;
         }
 
-        final ArmeriaChannel channel = new ArmeriaChannel(
+        final List<ClientInterceptor> interceptors = options.get(GrpcClientOptions.INTERCEPTORS);
+        final Channel channel = ClientInterceptors.intercept(new ArmeriaChannel(
                 newParams,
                 httpClient,
                 meterRegistry(),
                 scheme.sessionProtocol(),
                 serializationFormat,
                 jsonMarshaller,
-                simpleMethodNames);
-
+                simpleMethodNames), interceptors);
```

Comment:
Could we apply `ClientInterceptors.intercept` only when `interceptors` is non empty?
```java
if (!interceptors.isEmpty()) {
    ...
}
```